### DISCUSSION TITLE
Detekt CLI Type Resolution - How to get the classpath

### DISCUSSION THREAD

**Discussion by gilgoldzweig** [detekt/detekt#6550] [Q&A] [answered]
Hi, I'm working on a large Android project with many modules. 
We use detekt both as a pre-commit hook and a GitHub action. 
If I run detekt using the Gradle plugin, I'm able to use type resolution, but it comes at the cost of speed of setting up and Gradle Configuration. 

I want to try and attach it to the application's build process and generate the classpath ahead of time, both to use for commits and for speeding up the CI pipe. 
Maybe print it to a file that can be pushed to git; that way, it can also be used for CI.

Does anyone know how I can generate that ahead of time?

Regardless of getting it ahead of time, I think the docs could be a bit more helpful on how to get the classpath, as it's not so clear.
I don't mind updating the docs and push a PR after I'll get more info

**Comment by cortinico** [comment, marked as answer]
The whole point of using Detekt Gradle Plugin is so you don't have to create the classpath manually.

In a sense, it's like if you where to ask how you can use the `javac` command to compile the `MainActivity.java` file, I don't believe that https://developer.android.com/ tells you how to do it.

To answer your questions I would do the following:
1. Clone the `detekt` repo
2. Edit the `detekt-gradle-plugin` to print the classpath on console. It should be a colon separated list of jars/folders.
3. Use it in your invocation of detekt CLI

Also I don't think we should update the docs to suggest this
